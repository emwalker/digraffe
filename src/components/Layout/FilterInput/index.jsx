// @flow
import React, { Component } from 'react'

type Props = {
  onEnter?: ?Function,
  value: string,
}

type State = {
  value: string,
}

class FilterInput extends Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      value: props.value,
    }
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ value: nextProps.value })
  }

  onChange = (event) => {
    this.setState({ value: event.target.value })
  }

  onEnter = (string) => {
    if (this.props.onEnter)
      this.props.onEnter(string)
  }

  onKeyPress = (event) => {
    if (event.key === 'Enter')
      this.onEnter(event.target.value)
  }

  render = () => (
    <input
      aria-label="Filter input"
      className="form-control"
      onChange={this.onChange}
      onKeyPress={this.onKeyPress}
      placeholder="Search"
      size="40"
      type="text"
      value={this.state.value}
    />
  )
}

export default FilterInput
