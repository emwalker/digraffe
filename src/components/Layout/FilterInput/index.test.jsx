import React from 'react'
import {shallow} from 'enzyme'

import FilterInput from './index'

const props = {
  onEnter: jest.fn(),
  value: 'some search phrase',
}

describe('<FilterInput />', () => {
  const wrapper = shallow(<FilterInput {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
