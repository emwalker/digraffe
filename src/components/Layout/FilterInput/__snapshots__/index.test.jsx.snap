// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterInput /> renders 1`] = `
<input
  aria-label="Filter input"
  className="form-control"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Search"
  size="40"
  type="text"
  value="some search phrase"
/>
`;
